exporter/exporterhelper/internal/bounded_memory_queue_test.go

```diff
@@ -257,23 +251,47 @@ func (s *consumerState) assertConsumed(expected map[string]bool) {
 }
 
 func TestZeroSizeWithConsumers(t *testing.T) {
-	q := NewBoundedMemoryQueue(0, 1)
+	q := NewBoundedMemoryQueue(0)
 
-	err := q.Start(context.Background(), componenttest.NewNopHost(), newNopQueueSettings())
+	err := q.Start(context.Background(), componenttest.NewNopHost())
+	stopWG := sync.WaitGroup{}
+	startConsumers(q, 1, &stopWG, func(item any) {})
 	assert.NoError(t, err)
 
 	assert.NoError(t, q.Offer(context.Background(), newStringRequest("a"))) // in process
 
 	assert.NoError(t, q.Shutdown(context.Background()))
+	stopWG.Wait()
 }
 
 func TestZeroSizeNoConsumers(t *testing.T) {
-	q := NewBoundedMemoryQueue(0, 0)
+	q := NewBoundedMemoryQueue(0)
 
-	err := q.Start(context.Background(), componenttest.NewNopHost(), newNopQueueSettings())
+	err := q.Start(context.Background(), componenttest.NewNopHost())
 	assert.NoError(t, err)
 
 	assert.ErrorIs(t, q.Offer(context.Background(), newStringRequest("a")), ErrQueueIsFull) // in process
 
 	assert.NoError(t, q.Shutdown(context.Background()))
 }
+
+func startConsumers(queue Queue, numConsumers int, stopWG *sync.WaitGroup, consume func(req any)) {
```

Comment:
What about, to reduce duplicate code, we make this a "Consumers" struct which has a "Shutdown" func. So that we can use this in the queue_sender as well?

Reply:
Sounds good. I applied your suggestion, PTAL